group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -147,6 +147,11 @@ public static class DeadlineAndEpoch {
      */
     private final TimelineHashMap<String, Integer> subscribedTopicNames;
 
+    /**
+     * Partition assignments per topic.
+     */
+    private final TimelineHashMap<Uuid, TimelineHashMap<Integer, String>> partitionAssignments;
```

Comment:
nit: Could we please move it next to `targetAssignment` as they go together? It would also be great if we could add a bit more details to the javadoc.

Reply:
I also wonder whether we could include `targetAssignment` in the same in order to stress that this is the counter part of `targetAssignment`.

Reply:
like targetPartitionAssignments?

Reply:
How about invertedTargetAssiments or something similar?